.dotnet/src/Custom/Batch/BatchClient.Protocol.cs

```diff
@@ -10,33 +11,72 @@ public partial class BatchClient
     /// <summary>
     /// [Protocol Method] Creates and executes a batch from an uploaded file of requests
     /// </summary>
+    /// <param name="returnWhen"> <see cref="ReturnWhen.Completed"/> if the
+    /// method should return when the service has finished running the 
+    /// operation, or <see cref="ReturnWhen.Started"/> if it should return 
+    /// after the operation has been created but may not have completed 
+    /// processing. </param>
     /// <param name="content"> The content to send as the body of the request. </param>
     /// <param name="options"> The request options, which can override default behaviors of the client pipeline on a per-call basis. </param>
     /// <exception cref="ArgumentNullException"> <paramref name="content"/> is null. </exception>
     /// <exception cref="ClientResultException"> Service returned a non-success status code. </exception>
-    /// <returns> The response returned from the service. </returns>
-    public virtual async Task<ClientResult> CreateBatchAsync(BinaryContent content, RequestOptions options = null)
+    /// <returns> A <see cref="BatchOperation"/> that can be used to wait for 
+    /// the operation to complete, or cancel the operation. </returns>
+    public virtual async Task<BatchOperation> CreateBatchAsync(ReturnWhen returnWhen, BinaryContent content, RequestOptions options = null)
     {
         Argument.AssertNotNull(content, nameof(content));
 
         using PipelineMessage message = CreateCreateBatchRequest(content, options);
-        return ClientResult.FromResponse(await _pipeline.ProcessMessageAsync(message, options).ConfigureAwait(false));
+
+        PipelineResponse response = await _pipeline.ProcessMessageAsync(message, options).ConfigureAwait(false);
+
+        using JsonDocument doc = JsonDocument.Parse(response.Content);
+        string batchId = doc.RootElement.GetProperty("id"u8).GetString();
+        string status = doc.RootElement.GetProperty("status"u8).GetString();
+
+        BatchOperation operation = new BatchOperation(_pipeline, _endpoint, batchId, status, response);
```

Comment:
Can we add a helper for lines 38-43? e.g.
```csharp
public class OperationResult {
    public OperationResult WaitFor(ReturnWhen returnWhen, RequestOptions? options) {
        if (returnWhen == ReturnWhen.Started) return operation;
        await operation.WaitForCompletionAsync(options?.CancellationToken ?? default).ConfigureAwait(false);
        return operation;
    }
}
```

this way this method body would be:
```csharp
BatchOperation operation = new BatchOperation(_pipeline, _endpoint, batchId, status, response);
return operation.WaitFor(returnWhen, options);
```

Maybe we can even remove the WaitForCompletion methods and simply ask users to use WaitFor(ReturnWhen.Completed);

... at which point, maybe we rename ReturnWhen

Reply:
Interesting!  Let me think more about this.  This would be a bigger move away from Azure.Core precedent, but I'm not opposed at first glance.

Reply:
Per offline discussion, we'll add an internal helper to these LRO types to simplify the client implementation.  We can always make it public later on if we see value in that.  We won't change the SCM base types due to a desire to preserve alignment across SCM and Azure.Core patterns here.

Reply:
Addressed.